tests/test_monitorcontrol.py

```diff
@@ -152,3 +165,52 @@ def test_power_mode(
     elif isinstance(expected, type(Exception)):
         with pytest.raises(expected):
             monitor.set_power_mode(mode)
+
+
+# ASUS VG27A when set to a mode that doesnt exist returnd analog1 (0x1)
+@pytest.mark.skipif(
+    USE_ATTACHED_MONITORS, reason="Real monitors dont support all input types"
+)
+@pytest.mark.parametrize(
+    "mode, expected",
+    [
+        ("ANALOG1", 0x01),
+        ("ANALOG2", 0x02),
+        ("DVI1", 0x03),
+        ("DVI2", 0x04),
+        ("COMPOSITE1", 0x05),
+        ("COMPOSITE2", 0x06),
+        ("SVIDEO1", 0x07),
+        ("SVIDEO2", 0x08),
+        ("TUNER1", 0x09),
+        ("TUNER2", 0x0A),
+        ("TUNER3", 0x0B),
+        ("CMPONENT1", 0x0C),
+        ("CMPONENT2", 0x0D),
+        ("CMPONENT3", 0x0E),
+        ("DP1", 0x0F),
+        ("DP2", 0x10),
+        ("HDMI1", 0x11),
+        ("HDMI2", 0x12),
+    ],
+)
+def test_input_source(
+    monitor: Monitor,
+    mode: Union[str, int],
+    expected: Union[int, Type[Exception]],
+):
+    if isinstance(expected, (int, str)):
+        monitor.set_input_source(mode)
+        read_source = monitor.get_input_source()
+        assert read_source == mode
+
+
+def test_get_vcp_capabilities(monitor: Monitor):
+    monitors_dict = monitor.get_vcp_capabilities()
+    model = monitors_dict["model"]
+    inputs = monitors_dict["inputs"]
+    print(inputs)
+    if model != "ACER VG271U":
+        raise AssertionError("Could not parse model")
+    if set(inputs) ^ set(["DP1", "HDMI1", "HDMI2"]) != set():
```

Comment:
I think this is just comparing two sets for equality?  Could be shortened to this.
```suggestion
    if set(inputs) != {"DP1", "HDMI1", "HDMI2"}:
```

Reply:
TIL about sets. Thanks.